xpk.py

```diff
@@ -3257,6 +3257,12 @@ def get_main_container(args, system, docker_image, resource_type) -> str:
     command = ('TPU_STDERR_LOG_LEVEL=0 TPU_MIN_LOG_LEVEL=0 TF_CPP_MIN_LOG_LEVEL=0'
                f' TPU_VMODULE=real_program_continuator=1 {args.command}')
 
+  xpk_return_user_exit_code = ''
+  if args.restart_on_user_code_failure:
+    if args.max_restarts <= 0:
+      xpk_print(f'Warning: --max-restarts, is set to {args.max_restarts}. Will not restart on user failure.')
+    xpk_return_user_exit_code = 'exit $EXIT_CODE'
```

Comment:
Also, this entire block seems to be repetitive at line 3746. Can we modify this somehow to reuse the code?

Reply:
Its a good future improvement, but not part of this PR. Agreed on the need due to Pathways, GPU, TPU, CPU diverging code flows.